cli/utilities/packages.py

```diff
@@ -0,0 +1,181 @@
+        self.manager = manager
+
+    def info(self, pkg=None):
+        """show a package information
+
+        Args:
+            pkg (str): package name
+        """
+        cmd = f"{self.manager} info"
+        if pkg:
+            cmd += f" {pkg}"
+
+        stdout, stderr = self.execute(sudo=True, cmd=cmd)
+        if stderr:
+            return False
+        return stdout
+
+    def list(self, pkg=None):
+        """list a package installed
+
+        Args:
+            pkg (str): package name
+        """
+        cmd = f"{self.manager} list "
+        if pkg:
+            cmd += f" {pkg}"
+        stdout, stderr = self.execute(sudo=True, cmd=cmd)
+        if stderr:
+            return False
+        return stdout
+
+    def install(self, pkg):
+        """install a package or packages
+
+        Args:
+            pkg (str): package need to be installed
+        """
+        cmd = f"{self.manager} install -y {pkg}"
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def upgrade(self, pkg):
+        """upgrade a package or packages
+
+        Args:
+            pkg (str): package need to be installed
+        """
+        cmd = f"{self.manager} upgrade -y {pkg}"
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def add_repo(self, repo, nogpgcheck=False):
+        """enable repos with config-manager command
+
+        Args:
+            repo (str): repo to be enabled
+        """
+        cmd = f"yum-config-manager --add-repo {repo}"
+        if nogpgcheck:
+            cmd += " --nogpgcheck"
+
+        return self.execute(sudo=True, cmd=cmd)
+
+    def clean(self):
+        """clean repos"""
+        cmd = f"{self.manager} clean all"
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+
+class SubscriptionManager(Cli):
+    """This module provides CLI interface for RH Subscription Manager."""
+
+    def __init__(self, nodes):
+        super(SubscriptionManager, self).__init__(nodes)
+
+        self.base_cmd = "subscription-manager"
+        self.repos = Repos(nodes, self.base_cmd)
+
+    def register(self, username, password, serverurl=None, baseurl=None, force=False):
+        """Register system to the Customer Portal or another subscription management service.
+
+        Args:
+            username (str): username to use when authorizing against the server
+            password (str): password to use when authorizing against the server
+            serverurl (str): server URL
+            baseurl (str): base URL for content
+            force (bool): register the system even if it is already registered
+        """
+        cmd = f"{self.base_cmd} register --username {username} --password {password}"
+        if serverurl:
+            cmd += f" --serverurl {serverurl}"
+
+        if baseurl:
+            cmd += f" --baseurl {baseurl}"
+
+        if force:
+            cmd += " --force"
+
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
```

Comment:
IMO, we shouldn't set `long_running=True`... We need to ensure it passes and raise on failure. This being a utility and not something we are going to test... we can add intelligence here.

The implemented approach is useful when we are testing the functionality of it.

Reply:
Thanks @psathyan, updated all related methods